src/subscriber.ts

```diff
@@ -268,6 +273,26 @@ export class Subscriber extends EventEmitter {
     this._subscription = subscription;
     this.setOptions(options);
   }
+
+  /**
+   * Sets our subscription properties from the first incoming message.
```

Comment:
`exactlyOnceDeliveryEnabled` can be changed any time, so we should check every message, not just the first one. You'd need some sort of throttle on the error, ofc.